sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -805,10 +807,10 @@ class Analyzer(
            * that this rule cannot handle. When that is the case, there must be another rule
            * that resolves these conflicts. Otherwise, the analysis will fail.
            */
-          right
+          (right, AttributeMap.empty[Attribute])
```

Comment:
Since I don't want to build an empty object per call, how about defining an empty map in this class field? e.g.,
```
  object ResolveReferences extends Rule[LogicalPlan] {

    private val emptyAttrMap = new AttributeMap[Attribute](Map.empty)
```

Reply:
just commented the same meanwhile :) sorry, I saw your comments only now :)